src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -2079,28 +2079,24 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
         } else if ((right_is_int || left_is_int) && op == ASR::binopType::Mul) {
             // string repeat
             int64_t left_int = 0, right_int = 0, dest_len = 0;
-            if (right_is_int) {
+            dest_type = ASRUtils::TYPE(ASR::make_Character_t(al,
+                            loc, 1, 1, nullptr));
+            if (right_is_int && ASRUtils::expr_value(right) != nullptr) {
                 ASR::Character_t *left_type2 = ASR::down_cast<ASR::Character_t>(
                     ASRUtils::type_get_past_array(left_type));
                 LCOMPILERS_ASSERT(ASRUtils::extract_n_dims_from_ttype(left_type) == 0);
                 right_int = ASR::down_cast<ASR::IntegerConstant_t>(
                                                    ASRUtils::expr_value(right))->m_n;
                 dest_len = left_type2->m_len * right_int;
                 if (dest_len < 0) dest_len = 0;
-                dest_type = ASR::down_cast<ASR::ttype_t>(
-                        ASR::make_Character_t(al, loc, left_type2->m_kind,
-                        dest_len, nullptr));
-            } else if (left_is_int) {
+            } else if (left_is_int && ASRUtils::expr_value(left) != nullptr) {
                 ASR::Character_t *right_type2 = ASR::down_cast<ASR::Character_t>(
                     ASRUtils::type_get_past_array(right_type));
                 LCOMPILERS_ASSERT(ASRUtils::extract_n_dims_from_ttype(right_type) == 0);
                 left_int = ASR::down_cast<ASR::IntegerConstant_t>(
                                                    ASRUtils::expr_value(left))->m_n;
                 dest_len = right_type2->m_len * left_int;
                 if (dest_len < 0) dest_len = 0;
-                dest_type = ASR::down_cast<ASR::ttype_t>(
-                        ASR::make_Character_t(al, loc, right_type2->m_kind,
-                        dest_len, nullptr));
             }
```

Comment:
@Shaikh-Ubaid is this assertion here necessarily required?
https://github.com/lcompilers/lpython/blob/3ccdfdb6d6da234a31a0b28fb627691cc9d7e214/src/lpython/semantics/python_ast_to_asr.cpp#L2116

Is the value of `result` prone to bugs? If not, we might as well be able to completely remove this block as it serves no other purpose than to get `dest_len` for the above stated assertion.